six.py

```diff
@@ -909,12 +909,19 @@ def ensure_str(s, encoding='utf-8', errors='strict'):
       - `str` -> `str`
       - `bytes` -> decoded to `str`
     """
-    if not isinstance(s, (text_type, binary_type)):
-        raise TypeError("not expecting type '%s'" % type(s))
+    # Optimization: fast return for the common case. Improves performance
+    # by ~2-2.5x in Python 2 or 1.4-1.7x in Py3 for the case where
+    # s is a str. The uncommon case (unicode in 2, bytes in 3), ends up
+    # being around the same. The case that suffers is a subclass, or when
+    # an exception is thrown. Those are about 15-20% slower.
```

Comment:
The one thing I don't like about this comment is that it doesn't specify what the comparison is against. Looking at the commit, obviously, it's the previous state of the code. But the contextless reader doesn't know that.

Secondly, if you subclass `str`, good riddance!

Reply:
True... simplified to the first sentence.  all that text about performance comes from a microbenchmark.

There's no love for str subclasses here either, thankfully they've become way less common over the years. :)